### PR TITLE
Fix node setup in modules/icinga2.py and states/icinga2.py

### PR DESCRIPTION
**Pull request by devopsoffice-peter**
### What does this PR do?
We added an option to the `icinga2.node_setup` function to define `master_host` independent from `master` to avoid hiding features of the Icinga2 CLI.

This provides a fix for the situation where the Icinga master is not available via its FQDN, only its IP. Backwards compatibility is preserved with a default value from the master argument.

We also included the option in the icinga2.node_setup state, updated relevant documentation and fixed formatting issues (using the salt lint tools) in these files.

### What issues does this PR fix or reference?
N/A

### Previous Behavior
The icinga2.node_setup function passes the same argument for the `--master` and `--master_host` options of Icinga2 CLI.

### New Behavior
The icinga2.node_setup function takes two different arguments for the `--master` and `--master_host` options of Icinga2 CLI.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
No

### CODE REVIEW

File: salt/states/icinga2.py [hunk 1]
```diff
@@ -245,32 +242,37 @@ def request_cert(name, master, ticket, port="5665"):
...
-    ret["comment"] = "FAILED. Certificate requested failed with output: {0}".format(
+    ret["comment"] = "FAILED. Certificate requested failed with output: {}".format(
         cert_request["stdout"]
     )
     ret["result"] = False
     return ret
 
 
-def node_setup(name, master, ticket):
+def node_setup(name, master, ticket, master_host=None):
```
Comment (Akm0d) [inline salt/states/icinga2.py:255]: A unit test needs to be added for this new master_host functionality.  @waynew can often help with writing simple unittests during his Test Clinic on Tuesdays and Thursdays

Comment (devopsoffice-peter) [reply salt/states/icinga2.py:255]: I added unit tests for the new feature.

**Comment by Akm0d** [comment]
For the two pylint errors on napalm can we add a `# pylint: disable=no-name-in-module` above or on the affected lines?

File: salt/states/icinga2.py (hunk 1 above)
Comment (Akm0d) [reply salt/states/icinga2.py:255]: Awesome thank you!

**Comment by Akm0d** [comment]
Nevermind, @dmurphy18 Tells me that `napalm` and `napalm_base` are no longer needed, those lines where they are tried to be imported can be removed to solve the linter errors

**Comment by devopsoffice-peter** [comment]
> Nevermind, @dmurphy18 Tells me that `napalm` and `napalm_base` are no longer needed, those lines where they are tried to be imported can be removed to solve the linter errors

Great, should I commit the affected files with those removed? pre-commit will also reformat these files in that case, hope that's OK.

**Comment by dmurphy18** [comment]
@devopsoffice-peter Actually if I recall correctly, Mirceau told me that napalm_base is no longer supported, so napalm is all you should need.

**Comment by devopsoffice-peter** [comment]
> @devopsoffice-peter Actually if I recall correctly, Mirceau told me that napalm_base is no longer supported, so napalm is all you should need.

That should explain why was the previous commit correct but the last one unsuccessful. Also, the errors are not from files changed with this PR. I could add the pylint disable flag until the bgp and net runners get fixed to use napalm.

**Review by Akm0d — APPROVED** [review]

**Comment by Akm0d** [comment]
@devopsoffice-peter this all looks good!  You are amazing. I'll push this through the pipe into magnesium :+1:

**Comment by devopsoffice-peter** [comment]
@Akm0d I updated the branch but two checks are still failing.

**Comment by Akm0d** [comment]
If the failures are unrelated to your changes then periodically update the branch until it passes

**Comment by devopsoffice-peter** [comment]
@Akm0d Thanks for the hint, it finally succeeded.

**Review by Akm0d — APPROVED** [review]

**Review by dmurphy18 — CHANGES_REQUESTED** [review]
@devopsoffice-peter Are you still interested in this PR ?, if unresponsive the PR shall be closed after several days.
Test cases need addressing as well as resolving conflicts

**Comment by dmurphy18** [comment]
Closing due to inactivity. Please let me know if I need to re-open the PR, or feel free to submit a new PR against the master branch with the feedback submitted here.

**Pull request closed without merging**